dotCMS/src/main/java/com/dotcms/security/multipart/IllegalTraversalFilePathValidator.java

```diff
@@ -12,10 +12,29 @@ public class IllegalTraversalFilePathValidator implements SecureFileValidator {
     @Override
     public void validate(final String fileName) {
 
-        if (null == fileName || fileName.indexOf("/") != -1 || fileName.indexOf("\\") != -1 || fileName.indexOf("..") != -1) {
+        if (null == fileName || containsPathSeparators(fileName) || containsPathTraversal(fileName)) {
 
             SecurityLogger.logInfo(this.getClass(), "The filename: '" + fileName + "' is invalid");
             throw new IllegalArgumentException("Illegal Multipart Request");
         }
     }
+
+    /**
+     * Checks if the filename contains path separator characters
+     */
+    private boolean containsPathSeparators(final String fileName) {
+        return fileName.contains("/") || fileName.contains("\\");
+    }
+
+    /**
+     * Checks for actual path traversal patterns, not just any occurrence of ".."
+     * Allows filenames like "kek .. lol.png" but blocks "../../file.png"
+     */
+    private boolean containsPathTraversal(final String fileName) {
```

Comment:
Does it work if there are more than 2 dots? for example: `lol.....png`, `test...again...png`, `.....png`?